lib/coverband.rb

```diff
@@ -66,10 +66,16 @@ def self.configuration
 
   def self.start
     Coverband::Collectors::Coverage.instance
-    # TODO: Railtie sets up at_exit after forks, via middleware, perhaps this hsould be
+    # TODO: Railtie sets up at_exit after forks, via middleware, perhaps this should be
     # added if not rails or if rails but not rackserverrunning
-    AtExit.register
-    Background.start if configuration.background_reporting_enabled && !RackServerCheck.running?
+    AtExit.register unless tasks_to_ignore?
+    Background.start if configuration.background_reporting_enabled && !RackServerCheck.running? && !tasks_to_ignore?
+  end
+
+  def self.tasks_to_ignore?
+    (defined?(Rake) &&
+    Rake.respond_to?(:application) &&
+    (Rake&.application&.top_level_tasks || []).any? { |task| Coverband::Configuration::IGNORE_TASKS.include?(task) })
```

Comment:
We could also assume that top_level_tasks is always present since before 2009: https://github.com/ruby/rake/commit/2de58bc9e1b1ab89564f160c58f768cefa736e70#diff-26729347a9c3b31c5c9f4305f965d8e3

That could take this down to?
```ruby
defined?(Rake) &&
  (Coverband::Configuration::IGNORE_TASKS & Array(Rake.application.top_level_tasks)).any?
```

Reply:
Wow, nice digging in Rake source... This is way simpler, updating.

Reply:
hmm when trying to test removing `assets:clean` on heroku this blew up...

I think when you run rake these are all there but when you run other Ruby processes there is stuff that tries to patch Rake or it is not fully loaded. Adding the checks back.

```
 ! Unable to load application: NoMethodError: undefined method `application' for Rake:Module
2019-05-29T14:20:39.755959+00:00 app[web.1]: bundler: failed to load command: puma (/app/vendor/bundle/ruby/2.6.0/bin/puma)
2019-05-29T14:20:39.756036+00:00 app[web.1]: NoMethodError: undefined method `application' for Rake:Module
2019-05-29T14:20:39.756039+00:00 app[web.1]: /app/vendor/bundle/ruby/2.6.0/bundler/gems/coverband-5c6f3d14b3b6/lib/coverband.rb:77:in `tasks_to_ignore?'
2019-05-29T14:20:39.756040+00:00 app[web.1]: /app/vendor/bundle/ruby/2.6.0/bundler/gems/coverband-5c6f3d14b3b6/lib/coverband.rb:71:in `start'
2019-05-29T14:20:39.756042+00:00 app[web.1]: /app/vendor/bundle/ruby/2.6.0/bundler/gems/coverband-5c6f3d14b3b6/lib/coverband.rb:99:in `<module:Coverband>'
2019-05-29T14:20:39.756043+00:00 app[web.1]: /app/vendor/bundle/ruby/2.6.0/bundler/gems/coverband-5c6f3d14b3b6/lib/coverband.rb:32:in `<main>'
2019-05-29T14:20:39.756045+00:00 app[web.1]: /app/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
2019-05-29T14:20:39.756047+00:00 app[web.1]: /app/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
2019-05-29T14:20:39.756048+00:00 app[web.1]: /app/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:83:in `register'
2019-05-29T14:20:39.756050+00:00 app[web.1]: /app/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
2019-05-29T14:20:39.756052+00:00 app[web.1]: /app/vendor/bundle/ruby/2.6.0/gems/bootsnap-1.4.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
```